File: chapter12/control_flow_1.ex
defmodule FizzBuzz do
  def upto(n) when n > 0, do: Enum.map(1..n, &fizzbuzz/1)

  defp fizzbuzz(n) do
    case {rem(n, 3), rem(n, 5)} do
      {0, 0} -> "FizzBuzz"
      {0, _} -> "Fizz"
      {_, 0} -> "Buzz"
      _ -> n
    end
  end
end

File: chapter12/control_flow_1.clj
(ns control-flow-1)

(defn- fizzbuzz [n]
  (let [fizz? (zero? (rem n 3))
        buzz? (zero? (rem n 5))]
    (cond
      (and fizz? buzz?) "FizzBuzz"
      fizz? "Fizz"
      buzz? "Buzz"
      :else n)))

(defn upto [n]
  (when (pos? n) (map fizzbuzz (range 1 (inc n)))))
